### PR TITLE
MEN-2927: Deploy a system update tutorial based on Raspbian image.

### PR DESCRIPTION
**Pull request by lluiscampos**
Repurposed a bit the whole guide to take the most advantage of the new
image and the snapshots feature: the user will now use two devices, one
serving as a golden device and the other one with a remote device role.

This PR depends on #857. I am launching it now to gather early feedback on the approach and the wording. Please @estenberg have a close look!

Very much based on [this commit](https://github.com/kacf/mender-docs/commit/f7ebcc237ce68d004cea99411858778cafddb091#diff-370bd976af8115d3c193b4c61a65e9bb) by @kacf. But I squashed the commits for cleaner Git history.

### CODE REVIEW

File: 08.Downloads/docs.md [hunk 1]
```diff
@@ -4,6 +4,47 @@ taxonomy:
...
+  devices without Mender running already.
+* Mender Artifacts (`*.mender`): Upload them to the Mender server in order to
+  deploy new root file systems to devices already running Mender and registered
+  with the server.
+
+Mender provides images based on two different operating systems:
+
+* One image for **Raspberry Pi 3**, which is based on the [Raspbian Linux
+  distribution](https://www.raspberrypi.org/downloads/raspbian/?target=_blank)
+* One demo image for **Beaglebone Black**, which is based on a minimal [Yocto
```
Comment (estenberg) [inline 08.Downloads/docs.md]: I would drop this (BeagleBone), it's a completely different beast.

Rather link to Mender Hub for community-grade images (I think Debian family of integrations would be the most relevant as we should get OOTB images there).

Also perhaps a note that Yocto images is not provided as the workflow in Yocto Project is to build from source. Then link to Yocto Project Mender Hub category.

File: 01.Getting-started/02.On-premise-installation/04.Deploy-a-system-update-demo/docs.md [hunk 2]
```diff
@@ -183,71 +273,89 @@ Which information is collected about devices is fully configurable; see the docu
...
+
+In this section, we assume that `$IP_OF_MENDER_CLIENT` is the IP address of your
+your device.
+
+!!! If you are using `bash`, you can set a variable before running the command
+!!! below, for example `IP_OF_MENDER_CLIENT="192.168.10.2"`. If you don't know
+!!! it, run `ip addr` on your device.
+
+This section will create a Mender Artifact from a running device using the
+snapshots feature of Mender. See [Inventory](../../artifacts/snapshots) to learn
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply 01.Getting-started/02.On-premise-installation/04.Deploy-a-system-update-demo/docs.md]: Typo. Fixed.

**Comment by lluiscampos** [comment]
> I would drop this (BeagleBone), it's a completely different beast.

@estenberg So I am very much in favor for this, dropping the Yocto based image from both the `Downloads` page and the full `Deploy a system update demo` tutorial.

A Yocto based image is not suitable for showcasing the Snapshot feature or in general creating Artifact from run time modification (in my first proposal for this guide, in case of Yocto the update itself is of very low value). Furthermore, it will simplify greatly the guide if we can assume we are running Debian based OS. The only downside is that until we get BBB in place with mender-convert, we are only supporting one board in this tutorial, but IMO this is an acceptable compromise.

I am reworking this to drop Yocto demo image and Yocto mentions in the guide, I hope I didn't misunderstand!

> Rather link to Mender Hub for community-grade images (I think Debian family of integrations would be the most relevant as we should get OOTB images there).
> Also perhaps a note that Yocto images is not provided as the workflow in Yocto Project is to build from source. Then link to Yocto Project Mender Hub category.

Yep, I will add mentions to these.

File: 01.Getting-started/02.On-premise-installation/04.Deploy-a-system-update-demo/docs.md [hunk 3]
```diff
@@ -183,71 +273,89 @@ Which information is collected about devices is fully configurable; see the docu
...
-## Prepare the Mender Artifact to update to
+!!! If your device does not show up for authorization in the UI, you need to
+!!! diagnose what went wrong. Most commonly this is due to problems with the
+!!! network. You can test if your workstation can reach the device by trying to ping
+!!! it, e.g. with `ping 192.168.10.2` (replace with the IP address of your
+!!! device). If you can reach the device, you can ssh into it, e.g. `ssh root@192.168.10.2`.
+!!! Otherwise, if you have a serial cable, you can log in to the device to
+!!! diagnose. The `root` user is present and has an empty password in this test
+!!! image. Check the log output from Mender with `journalctl -u mender-client`
+!!! or `journalctl -u mender`. If you get stuck, please feel free to reach out
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply 01.Getting-started/02.On-premise-installation/04.Deploy-a-system-update-demo/docs.md]: I overlooked this text, yes. It needs rework.

I also realized that the serial console is not an option neither, so I will hint SSH or HDMI+keyboard. Also earlier, on the mender `setup part`, I will explicitly mention the possible access methods.

File: 01.Getting-started/02.On-premise-installation/04.Deploy-a-system-update-demo/docs.md [hunk 4]
```diff
@@ -183,71 +273,89 @@ Which information is collected about devices is fully configurable; see the docu
...
 
-Then check the contents of the file
+## Generate an Artifact from the golden device
+
+In this section, we assume that `$IP_OF_MENDER_CLIENT` is the IP address of your
+your device.
+
+!!! If you are using `bash`, you can set a variable before running the command
+!!! below, for example `IP_OF_MENDER_CLIENT="192.168.10.2"`. If you don't know
+!!! it, run `ip addr` on your device.
```
[the remark this answers is not in the document]
Comment (lluiscampos) [reply 01.Getting-started/02.On-premise-installation/04.Deploy-a-system-update-demo/docs.md]: Modified the text to use `hostname -I`, which is the recommended method in [Raspberry Pi Docs](https://www.raspberrypi.org/documentation/remote-access/ip-address.md).

File: 08.Downloads/docs.md [hunk 5]
```diff
@@ -4,6 +4,43 @@ taxonomy:
...
+* Image for **Raspberry Pi 3**, which is based on the [Raspbian Linux
+  distribution](https://www.raspberrypi.org/downloads/raspbian/?target=_blank)
+
+!! Note that we do not offer support for any of these images. Please [contact
+!! us](mailto:contact@mender.io) if you would like to discuss professional
+!! support.
+
+| Board            | OS                      | Disk image                                                               | Mender Artifact                                                                              |
+|------------------|-------------------------|--------------------------------------------------------------------------|----------------------------------------------------------------------------------------------|
+| Raspberry Pi 3   | Raspbian Buster Lite    | [raspbian-buster-lite-mender.img.xz][raspbian-buster-lite-mender.img.xz] | [raspbian-buster-lite-mender_release-1.mender][raspbian-buster-lite-mender_release-1.mender] |
```
Comment (lluiscampos) [inline 08.Downloads/docs.md]: I left the Artifact based on [this comment](https://tracker.mender.io/browse/MEN-2927?focusedCommentId=103390&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-103390). However the Artifact is not being updated when the image is deployed, so I created [MEN-3148](https://tracker.mender.io/browse/MEN-3148) for it

**Comment by lluiscampos** [comment]
> I am reworking this to drop Yocto demo image and Yocto mentions in the guide, I hope I didn't misunderstand!

Done in the last rework. Review is appreciated!

I decided to keep it in a separate commit, so that we can eventually pick it up again when we have BBB mender-converted image in place (for example, all the explanation on how to boot BBD from SD card)

File: 08.Downloads/docs.md (hunk 5 above)
Comment (estenberg) [reply 08.Downloads/docs.md]: My take is that I would not spend much time on this, and just get rid of the Artifacts. Snapshot feature is done and is a much better and complete way to test and understand what is going on. In fact, it is possibly confusing to some new users what those Artifacts are (again referring to a single default user journey).

I am not against doing it if you want to though, just think there is limited value. And will the network config work? I assume the tenant token and all that will be erased, resulting in a failed update?

Comment (estenberg) [reply 08.Downloads/docs.md]: One more comment, can we include the OS version here?

File: 08.Downloads/docs.md [hunk 6]
```diff
@@ -4,6 +4,43 @@ taxonomy:
...
+* Mender Artifacts (`*.mender`): Upload them to the Mender server in order to
+  deploy new root file systems to devices already running Mender and registered
+  with the server.
+
+Mender provides images based on the following distributions:
+
+* Image for **Raspberry Pi 3**, which is based on the [Raspbian Linux
+  distribution](https://www.raspberrypi.org/downloads/raspbian/?target=_blank)
+
+!! Note that we do not offer support for any of these images. Please [contact
```
Comment (estenberg) [inline 08.Downloads/docs.md]: Perhaps say "Note that we do not offer commercial support for these images. They are based on images supported by board manufacturers, like the Raspberry Pi Foundation, and provide the same software and configuration options as the original images. Please use the support resources available from the board manufacturer or [contact us](mailto:contact@mender.io) if you have any questions Mender integration."

**Comment by estenberg** [comment]
Nice job!

File: 08.Downloads/docs.md (hunk 6 above)
Comment (lluiscampos) [reply 08.Downloads/docs.md]: :+1:

File: 08.Downloads/docs.md (hunk 5 above)
Comment (lluiscampos) [reply 08.Downloads/docs.md]: > I assume the tenant token and all that will be erased, resulting in a failed update?

Yes, without modifications they will result in failed updates, mainly due to missing `mender.conf` file completely. As we are not guiding the user anymore on how to to this, I agree with you in just removing them. Done in [last commit](https://github.com/mendersoftware/mender-docs/pull/858/commits/03d8555bd88b17a6c2f4284ed59059343ff51e40).

> One more comment, can we include the OS version here?

Added :+1:

**Comment by lluiscampos** [comment]
@kacf I think this is ready. In the last push I fixed some anchor links that were broken and couple of changes following @estenberg's comments. Please have a final review, so that we have it ready once #857 is merged.

File: .ispell_english [hunk 7]
```diff
@@ -641,6 +641,7 @@ dtc
 DTS
 DULG
 Dulin
+dumpe
```
Comment (kacf) [inline .ispell_english]: Lol, it took me some minutes to figure out where this word appears!

`dumpe2fs`! Of course!

**Pull request merged**